Removed fixed string from HelpFormatter

cod-sushi opened the pull request:
Removed fixed strings `"Commands:"` and help page ending note(`"Type ?help command for more info on a command ..."`)
and added properties modify these strings.
default behavior is not changed. fix #1886

I wrote these codes referencing code about `Bot.command_prefix` .

I added comment about added properties too.(for document)
I apologize if there are any mistakes in my English and document writing.

Rapptz commented on discord/ext/commands/formatter.py:
```diff
@@ -142,11 +142,31 @@ class HelpFormatter:
         self.width = width
         self.show_hidden = show_hidden
         self.show_check_failure = show_check_failure
+        self.ending_note = ending_note
```
I feel like this shouldn't be an attribute but something that should be changed through inheritance and subclassing. This is why it's in a separate method, so you can do something like the following:

```py
class MyHelpFormatter(commands.HelpFormatter):
    def get_ending_node(self):
      return '...'
```

I do agree with the `commands_heading` though so I think that's good.

cod-sushi commented:
Thanks for reviewing.
I removed `ending_note` property.
And, I notice there is "No Categoly", so I added property `no_categoly` to HelpFormatter to remove this fixed text.

Vexs commented:
Typo: spelled `category` not `categoly`

cod-sushi commented:
so embarrassed... I'll fix it soon. thank you vex :sob:

Rapptz commented:
Thanks for the pull request!
